[Maps] Added options to disable zoom, hide tool tips, widgets/overlays in embeddable maps

shahzad31 opened the pull request:
Fixes: #49871
## Summary

To achieve the skimmed down version to be used in Uptime app, 
Following configurable options are added into embeddable maps API

1. Disable Double click zooming 
2. Disable zoom(hide zoom control)
3. Hide widget overlays
4. Hide Panel header

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)
- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

elasticmachine commented:
Pinging @elastic/kibana-gis (Team:Geo)

elasticmachine commented:
Pinging @elastic/uptime (Team:uptime)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/9795/)
* Commit: 47b971b47739fe2cd1c0347714d80fa1b1cc12a5

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/9811/)
* Commit: 923bba22d74a80f77847548202bffa6f77566d38

nreese commented:
Thanks for opening a PR for #49871

Users can still zoom in by holding shift and dragging over an area. You may want to set `interactive: false` when initializing the map.

This PR introduces a lot of variables for the embeddable input:
* disableZoom
* disableTooltipControl
* hideToolbarOverlay
* hideWidgetOverlay

How about simplifying them into a single key, `isNonInteractiveMap`. When true, then the UI will hide all chrome, disable tooltips, and disable map interactions. How does that sound? So keep redux actions and selectors, but trigger from single embeddable input.

@thomasneirynck This PR hides attribution. Is that a good idea? I think attribution should never be allowed to be hidden. For uptime, none of the sources have attribution so it should not matter but I think there are legal requirements to show attribution for EMS layers using OSM data.

nreese commented:
The more I think about, maybe the redux state should be simplified as well and only have a single boolean `isNonInteractiveMap`. `GisMap` should get its state from redux connectors since its a connected component instead of passing state into the component with props. `addFilters` and `renderTooltipContent` are passed in as props because there are functions and can not be stored in redux state.

shahzad31 commented:
@nreese i agree about simplifying  redux state, though about first point, or accumulating all booleans into single booleans, i also thought of it. 

Why i opted for keeping all options separate to keep it more flexible, if different needs arises, i mean if someone wants to keep some toolbar or widget , or zooming. They can individually disable/enable those. 

Combining them will keep code clean, but it won't be as flexible. Maybe we can have a one redux action to pass whole object, something like 
`
mapOptions = {zoom: false, tooltip:false, etc } 
`
but that also doesn't look clean. 

Though just having isNonInteractiveMap does looks way more clean.

WDYT?

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/10141/)
* Commit: fa948dddcd5081d5f4aa0139aafa537df7481229

justinkambic commented:
> Why i opted for keeping all options separate to keep it more flexible

FWIW I'm +1 for flexibility too, combining the boolean flags seems like something a future consumer could run up against and want to change: "I want to enable tooltips but disable zooming".

bcamper commented:
I agree having some level of granularity over controls (zoom, pan, tooltips, legend etc.) makes sense.

As for attribution: generally, it's going to be required that we show it *somewhere*, but it does not necessarily need to be on the map itself, e.g. it could be nearby on the page, or sometimes apps that have other non-map components will group all attribution together in a single "Data sources" link. But, there's nuance there and concern that a client from another group wouldn't know the requirements, so we may want to be more conservative and not allow it to be disabled for now, wait and see if it comes up in practice.

justinkambic commented:
> we may want to be more conservative and not allow it to be disabled for now, wait and see if it comes up in practice.

👍

shahzad31 commented:
> @thomasneirynck This PR hides attribution. Is that a good idea? I think attribution should never be allowed to be hidden. For uptime, none of the sources have attribution so it should not matter but I think there are legal requirements to show attribution for EMS layers using OSM data.

@nreese @bcamper i don't understand the attribution part, i think

`  attributionControl: false,
`

was already there, i haven't changed that.

nreese commented:
> i don't understand the attribution part

This line in `{!this.props.hideWidgetOverlay && <WidgetOverlay />}` [gis_map/view.js](https://github.com/elastic/kibana/blob/fa948dddcd5081d5f4aa0139aafa537df7481229/x-pack/legacy/plugins/maps/public/connected_components/gis_map/view.js#L207)

`hideWidgetOverlay` should not hide the entire WidgetOverlay, just LayerControl and ViewControl but AttributionControl should always be visible and not be allowed to be turned off.

shahzad31 commented:
@nreese i have removed the option to hide attribution, i think we can keep widgetOverlay, it shouldn't be a problem. I have also refactored redux state and have moved all props to redux.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/10950/)
* Commit: b4cb029169d84330698e7bc2798114c4eab1c80a

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/11135/)
* Commit: ddb49c1590cb133526d043dbf686d3735198670f

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/11169/)
* Commit: 184a7e1f667aef57b0e9dc18273552ba57ed73f4

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/11256/)
* Commit: 6b68572366f98578700c095627af85cfbecb5eb2

nreese commented:
@elasticmachine merge upstream

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/11342/)
* Commit: e0ed451520c9addd3c005cd5a41e0705cdf4c23d
